### PR TITLE
Profinite and ExtrDisc are precoherent

### PR DESCRIPTION
**Pull request by riccardobrasca**
- [x] depends on: #9

### CODE REVIEW

File: Profinite/Epi.lean [hunk 1]
```diff
@@ -35,15 +35,6 @@ def pullback.snd : pullback f g ⟶ Y where
...
 
-open List in
-theorem effectiveEpiFamily_tfae {α : Type} [Fintype α] {B : Profinite.{u}}
-    (X : α → Profinite.{u})
-    (π : (a : α) → (X a ⟶ B)) :
-    TFAE [
-      EffectiveEpiFamily X π,
-      Epi (Limits.Sigma.desc π),
-      ∀ (b : B), ∃ (a : α) (x : X a), π a x = b
-    ] := by
```
Comment (kbuzzard) [inline Profinite/Epi.lean]: Did you mean to delete this code? Other than that, this looks fine.

File: ExtrDisc/Coherent.lean [hunk 2]
```diff
@@ -20,10 +21,35 @@ Supply the proof!
...
 
 namespace ExtrDisc
 
-instance : Precoherent ExtrDisc.{u} := sorry
+theorem _root_.CategoryTheory.EffectiveEpiFamily.toCompHaus {α : Type} [Fintype α] {B : ExtrDisc.{u}}
+    {X : α → ExtrDisc.{u}} {π : (a : α) → (X a ⟶ B)} (H : EffectiveEpiFamily X π) :
+    EffectiveEpiFamily (fun a => toCompHaus.obj (X a) ) (fun a => toCompHaus.map (π a)) := by 
+  refine' ((CompHaus.effectiveEpiFamily_tfae _ _).out 0 2).2 (fun b => _)
+  exact (((effectiveEpiFamily_tfae _ _).out 0 2).1 H : ∀ _, ∃ _, _) _
+
```
Comment (kbuzzard) [inline ExtrDisc/Coherent.lean:29]:
```suggestion
```

File: Profinite/Epi.lean (hunk 1 above)
Comment (riccardobrasca) [reply Profinite/Epi.lean]: Ops

**Comment by riccardobrasca** [comment]
This is ready for review.

**Comment by kbuzzard** [comment]
Thanks a lot!

**Pull request merged**